src/drivers/amd/renoir/ipc.c

```diff
@@ -91,6 +93,15 @@ static void irq_handler(void *arg)
 	if (status) {
 		/* Interrupt source */
 		if (sof_ipc_host_status()) {
+			lock = io_reg_read(PU_REGISTER_BASE + ACP_AXI2DAGB_SEM_0);
+			while (lock) {
+				lock = io_reg_read(PU_REGISTER_BASE + ACP_AXI2DAGB_SEM_0);
+				if (!delay_cnt) {
+					break;
+					return;
```

Comment:
unreachable code

Reply:
Added proper code and addressed the review comments